aggregator/src/datastore.rs

```diff
@@ -3358,8 +3315,8 @@ ORDER BY id DESC
                         FROM collect_jobs
                         JOIN aggregation_jobs
                             ON aggregation_jobs.task_id = collect_jobs.task_id
-                            AND (aggregation_jobs.batch_identifier = collect_jobs.batch_identifier
-                              OR aggregation_jobs.batch_interval <@ collect_jobs.batch_interval)
+                            AND (aggregation_jobs.batch_id = collect_jobs.batch_identifier
+                                OR aggregation_jobs.client_timestamp_interval <@ collect_jobs.batch_interval)
```

Comment:
I think this probably needs to use && as well, since we're dropping the assumption that the aggregation job's timestamp interval is minimally sized.